workers/unity/Packages/com.improbable.gdk.transformsynchronization/TransformUtils.cs

```diff
@@ -11,51 +11,146 @@ public static bool HasChanged(Coordinates a, Coordinates b)
         }
 
         // Checking for no change, so exact equality is fine
-        public static bool HasChanged(Location a, Location b)
+        public static bool HasChanged(CompressedQuaternion a, CompressedQuaternion b)
         {
-            return a.X != b.X || a.Y != b.Y || a.Z != b.Z;
+            return a.Data != b.Data;
         }
 
         // Checking for no change, so exact equality is fine
-        public static bool HasChanged(Quaternion a, Quaternion b)
+        public static bool HasChanged(FixedPointVector3 a, FixedPointVector3 b)
+        {
+            return a.X != b.X || a.Y != b.Y || a.Z != b.Z;
+        }
+
+        // Decompress from uint32 to Unity Quaternion
+        public static UnityEngine.Quaternion ToUnityQuaternion(this CompressedQuaternion quaternion)
+        {
+            var compressedValue = quaternion.Data;
+            var q = new float[4];
+
+            const uint mask = (1u << 9) - 1u;
+
+            int largestIndex = (int) (compressedValue >> 30);
+            float sumSquares = 0;
+            for (var i = 3; i >= 0; --i)
+            {
+                if (i != largestIndex)
+                {
+                    uint magnitude = compressedValue & mask;
+                    uint signBit = (compressedValue >> 9) & 0x1;
```

Comment:
Is the reason the mask is 9 bits to add room for a sign bit then? Can't the mapping from float to fixed just encode this in the main 10 bits?

Reply:
This seems like it just adds floating point comparisons and multiplications.

Reply:
I'll leave this as-is for now to get the changes in, and raise a new ticket for further optimisations

Reply:
It's not for optimisation so much as it seems odd.